models/issue_dependency.go

```diff
@@ -117,23 +132,14 @@ func RemoveIssueDependency(user *User, issue *Issue, dep *Issue, depType Depende
 }
 
 // Check if the dependency already exists
-func issueDepExists(e Engine, issueID int64, depID int64) (err error) {
-
-	// Check if the dependency exists
-	exists, err := e.Where("(issue_id = ? AND dependency_id = ?)", issueID, depID).Exist(&IssueDependency{})
-	if exists {
-		return ErrDependencyExists{issueID, depID}
-	}
-
-	// If not, check for circular dependencies
-	circular, err := e.Where("issue_id = ? AND dependency_id = ?", depID, issueID).Exist(&IssueDependency{})
-	if err != nil {
-		return err
-	}
-	if circular {
-		return ErrCircularDependency{issueID, depID}
-	}
+func issueDepExists(e Engine, issueID int64, depID int64) (exists bool, err error) {
+	exists, err = e.Where("(issue_id = ? AND dependency_id = ?)", issueID, depID).Exist(&IssueDependency{})
```

Comment:
Better now. But you don't need two functions as `issueDepExists` and `issueDepIsCircular` do the same thing, only with switched arguments. Keep only `issueDepExists`.

Little improvement:
```
func issueDepExists(e Engine, issueID int64, depID int64) (bool, error) {
	return e.Where("issue_id = ? AND dependency_id = ?", issueID, depID).Exist(&IssueDependency{})
}
```

Reply:
But I want to show an error message if the user is about to create a circular dependency, which is why I check for it.

Reply:
Sure, call `issueDepExists` twice, only switch arguments... Function `issueDepExists` and `issueDepIsCircular` does the exactly same thing.

Reply:
Right, I haven't thought of that.